geode-core/src/main/java/org/apache/geode/management/internal/configuration/mutators/ConfigurationManager.java

```diff
@@ -38,5 +38,5 @@
 
   void delete(T config, CacheConfig existing);
 
-  <R extends RuntimeCacheElement> List<R> list(T filterConfig, CacheConfig existing);
```

Comment:
This leaves the person implementing the ConfigurationManager unable to set the return type so that we still run the risk of getting unchecked cast or raw type warnings. Can we do something like this?:
```java
public interface ConfigurationManager<T extends CacheElement, R extends RuntimeCacheElement> {
  ...

  List<R> list(T filterConfig, CacheConfig existing);
}
```

Reply:
True, but this PR is not trying to solve this problem. It was that like that or has the same effect as before. We should try to solve this problem in the mobbing session next week.

Reply:
If I try to solve this problem here, the change set is going to be more than what it already has now.

Reply:
Okay, that makes sense. I just want to make sure that we address these issues before we grow the code too much as it is likely to get more complicated when we put this off for too long.
Should I open a chore for this?